tests/python/relax/test_transform_fuse_tir.py

```diff
@@ -2314,5 +2314,88 @@ def take(
     _check(Before, Before)
 
 
+def test_fuse_with_axis_separators():
+    @I.ir_module
+    class Before:
+        @T.prim_func(private=True)
+        def add(a: T.handle, b: T.handle, c: T.handle):
+            A = T.match_buffer(a, [T.int64(16), T.int64(32)], "float32", axis_separators=[1])
+            B = T.match_buffer(b, [T.int64(16), T.int64(32)], "float32", axis_separators=[1])
+            C = T.match_buffer(c, [T.int64(16), T.int64(32)], "float32", axis_separators=[1])
+
+            for iters in T.grid(T.int64(16), T.int64(32)):
+                with T.block("compute"):
+                    i, j = T.axis.remap("SS", iters)
+                    C[i, j] = A[i, j] + B[i, j]
+
+        @R.function(private=True)
+        def fused_function(
+            x: R.Tensor([T.int64(16), T.int64(32)], "float32"),
+            y: R.Tensor([T.int64(16), T.int64(32)], "float32"),
+            z: R.Tensor([T.int64(16), T.int64(32)], "float32"),
+        ) -> R.Tensor([T.int64(16), T.int64(32)], dtype="float32"):
+            R.func_attr({"Primitive": 1})
+            cls = Before
+            with R.dataflow():
+                w = R.call_tir(
+                    cls.add, [x, y], out_sinfo=R.Tensor([T.int64(16), T.int64(32)], "float32")
```

Comment:
Can we add a test case for incompatible usage of a single Relax var?  As currently written, we could have a single Relax variable that is used in two separate `R.call_tir` statements, where the function being called imposes different restrictions on it.  For example, if `x` were used in `cls.add1`, which requires `axis_separators=[1]`, and `cls.add2`, which requires `axis_separators=[]`.  We should be able to identify this case and raise an error when it occurs.

(Ideally, that should never happen, but this would be the last point at which we'd have enough information to catch this failure mode at compile-time.)

Reply:
I've added the test case to check possible inconsistencies.